src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -462,6 +462,47 @@ private void AssertNoPlaceholderReplacements()
             }
         }
 
+        private void AddPlaceholderReplacement(BoundValuePlaceholderBase placeholder, BoundExpression expression, VisitResult result)
+        {
+#if DEBUG
+            Debug.Assert(AreCloseEnough(placeholder.Type, result.RValueType.Type));
+#endif
+
+            _resultForPlaceholdersOpt ??= PooledDictionary<BoundValuePlaceholderBase, (BoundExpression Replacement, VisitResult Result)>.GetInstance();
+            _resultForPlaceholdersOpt.Add(placeholder, (expression, result));
+        }
+
+        private void RemovePlaceholderReplacement(BoundValuePlaceholderBase placeholder)
+        {
+            Debug.Assert(_resultForPlaceholdersOpt is { });
+            bool removed = _resultForPlaceholdersOpt.Remove(placeholder);
+            Debug.Assert(removed);
+        }
+
+        private static bool PlaceholderMustBeRegistered(BoundValuePlaceholderBase placeholder)
```

Comment:
This appears to only be used as an assert. Consider making this a conditional method and doing the assert directly.